server/src/main/kotlin/suwayomi/tachidesk/server/database/DBManager.kt

```diff
@@ -31,38 +33,87 @@ object DBManager {
     var db: Database? = null
         private set
 
+    @Volatile
+    private var hikariDataSource: HikariDataSource? = null
+
+    private fun createHikariDataSource(): HikariDataSource {
+        val applicationDirs = Injekt.get<ApplicationDirs>()
+        val config =
+            HikariConfig().apply {
+                when (serverConfig.databaseType.value) {
+                    DatabaseType.POSTGRESQL -> {
+                        jdbcUrl = "jdbc:${serverConfig.databaseUrl.value}"
+                        driverClassName = "org.postgresql.Driver"
+                        username = serverConfig.databaseUsername.value
+                        password = serverConfig.databasePassword.value
+                        // PostgreSQL specific optimizations
+                        addDataSourceProperty("cachePrepStmts", "true")
+                        addDataSourceProperty("prepStmtCacheSize", "25")
+                        addDataSourceProperty("prepStmtCacheSqlLimit", "256")
+                        addDataSourceProperty("useServerPrepStmts", "true")
+                    }
+                    DatabaseType.H2 -> {
+                        jdbcUrl = "jdbc:h2:${applicationDirs.dataRoot}/database"
+                        driverClassName = "org.h2.Driver"
+                        // H2 specific optimizations
+                        addDataSourceProperty("cachePrepStmts", "true")
+                        addDataSourceProperty("prepStmtCacheSize", "25")
+                        addDataSourceProperty("prepStmtCacheSqlLimit", "256")
+                    }
+                }
+
+                // Optimized for Raspberry Pi / Low memory environments
+                maximumPoolSize = 6 // Moderate pool for better concurrency
+                minimumIdle = 2 // Keep 2 idle connections for responsiveness
```

Comment:
What are the defaults? Would this be something that a user might want to configure to optimize performance on higher end machines?